frame/nomination-pools/src/lib.rs

```diff
@@ -2419,6 +2463,52 @@ impl<T: Config> Pallet<T> {
 		Ok(())
 	}
 
+	fn do_bond_extra(
+		who: T::AccountId,
+		other: Option<AccountIdLookupOf<T>>,
+		extra: BondExtra<BalanceOf<T>>,
+	) -> DispatchResult {
+		let other = other.map(T::Lookup::lookup).transpose()?;
+		let is_bonding_for = other.is_some();
+		let who = other.unwrap_or(who);
+		if is_bonding_for {
+			ensure!(
+				RewardClaimPermission::<T>::get(&who) == RewardClaim::Permissionless,
+				Error::<T>::DoesNotHavePermission
+			);
+			ensure!(extra == BondExtra::Rewards, Error::<T>::CannotBondFreeBalanceOther);
+		}
```

Comment:
```suggestion
		let other = T::Lookup::lookup(other)?;
		// either it is permissioned, or they have allowed it.
		if who != other {
			ensure!(matches!(RewardClaimPermission::<T>::get(&target), 
 RewardClaim::Permissionless), Error::<T>::DoesNotHavePermission);
 			ensure!(extra == BondExtra::Rewards, Error::<T>::CannotBondFreeBalanceOther);
		}
 );
```

Reply:
I actually realized that it is probably simpler without `Option` 🙈  although they are both the same.